src/model.h

```diff
@@ -118,47 +115,30 @@ struct RapidsModel : rapids::Model<RapidsSharedState> {
 
     output.finalize();
 
-    /* For running gpu treeshap
-     * Make sure that input and output buffers are in device memory
-     * for gpu treeshap */
-    if constexpr (rapids::IS_GPU_BUILD && IS_TREESHAP_BUILD) {
-      // boolean to check whether gpu treeshap will be executed
-      auto run_gpu_treeshap =
-          shared_state->check_output_name("treeshap_output");
-
-      if (run_gpu_treeshap) {
-        auto treeshap_output = get_output<float>(batch, "treeshap_output");
-        auto treeshap_output_buffer = rapids::Buffer<float>(
-            treeshap_output.data(), treeshap_output.size(),
-            treeshap_output.mem_type(), treeshap_output.device(),
-            treeshap_output.stream());
-
-        if (gpu_treeshap_model.has_value()) {
-          // Always copy input buffer to device memory
-          if (input_buffer.mem_type() == rapids::HostMemory) {
-            input_buffer = rapids::Buffer<float const>(
-                input_buffer, rapids::DeviceMemory, get_device_id());
-          }
-
-          // This will force a copy of the output buffer to device memory
-          if (input_buffer.mem_type() != treeshap_output_buffer.mem_type()) {
-            // Create output buffer in correct  location
-            treeshap_output_buffer = rapids::Buffer<float>(
-                treeshap_output.size(), input_buffer.mem_type(),
-                get_device_id(), get_stream());
-          }
-
+    // boolean to check whether gpu treeshap will be executed
+    auto run_treeshap =
+        shared_state->check_output_name("treeshap_output");
+    if (run_treeshap) {
+      auto treeshap_output = get_output<float>(batch, "treeshap_output");
+      auto treeshap_output_buffer = rapids::Buffer<float>(
+          treeshap_output.data(), treeshap_output.size(),
+          treeshap_output.mem_type(), treeshap_output.device(),
+          treeshap_output.stream());
+
+      if (gpu_treeshap_model.has_value() &&
+          input_buffer.mem_type() == rapids::DeviceMemory) {
+        if constexpr (rapids::IS_GPU_BUILD && IS_TREESHAP_BUILD) {
           // The shape of treeshap output is (, num_classes * (n_cols + 1))
           gpu_treeshap_model->predict(
               treeshap_output_buffer, input_buffer, samples, input.shape()[1]);
         }
-
-        if (treeshap_output_buffer.mem_type() != treeshap_output.mem_type()) {
-          rapids::copy(treeshap_output.buffer(), treeshap_output_buffer);
-        }
-
-        treeshap_output.finalize();
       }
+      else{
+          cpu_treeshap_model.predict(
```

Comment:
This should definitely be an optional guarded by an `if .has_value()`. We don't want to always load a treeshap model if it won't be used.